File: device/bluetooth/android/java/src/org/chromium/device/bluetooth/BluetoothLeScannerWrapper.java
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.device.bluetooth;

import android.Manifest;
import android.annotation.TargetApi;
import android.bluetooth.le.BluetoothLeScanner;
import android.bluetooth.le.ScanFilter;
import android.bluetooth.le.ScanCallback;
import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;

import org.chromium.base.CalledByNative;
import org.chromium.base.JNINamespace;
import org.chromium.base.Log;

import java.util.List;

/**
 * Wraps android.bluetooth.BluetoothLeScanner, pasing through to a provided
 * object. This indirection enables fake implementations when running tests.
 */
@JNINamespace("device")
@TargetApi(Build.VERSION_CODES.LOLLIPOP)
public class BluetoothLeScannerWrapper {
    private static final String TAG = "cr.Bluetooth";
    private final BluetoothLeScanner mScanner;

    public BluetoothLeScannerWrapper(BluetoothLeScanner scanner) {
        mScanner = scanner;
    }

    public void startScan(List<ScanFilter> filters, ScanSettings settings, ScanCallback callback)
        scanner.startScan(filters, settings, callback);
    }

    public void stopScan(ScanCallback callback) {
        scanner.stopScan(callback);
    }
}

File: device/bluetooth/android/bluetooth_le_scanner_wrapper.cc
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "device/bluetooth/android/bluetooth_le_scanner_wrapper.h"

#include "base/android/jni_android.h"
#include "jni/BluetoothLeScannerWrapper_jni.h"

using base::android::AttachCurrentThread;

namespace device {

// static
ScopedJavaLocalRef<jobject>
BluetoothLeScannerWrapper::CreateWithDefaultLeScanner() {
  return Java_BluetoothLeScannerWrapper_createWithDefaultLeScanner(
      AttachCurrentThread());
}

// static
bool BluetoothLeScannerWrapper::RegisterJNI(JNIEnv* env) {
  return RegisterNativesImpl(env);  // From BluetoothLeScannerWrapper_jni.h
}

}  // namespace device
